### PR TITLE
refactor: driver spec change

### PR DESCRIPTION
**Pull request by yuanyan**
*Before* submitting a pull request, please make sure the following is done...

1. Fork the repo and create your branch from `master`.
2. If you've added code that should be tested, add tests!
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).
5. Make sure your code lints (`npm run lint`) - we've done our best to make sure these rules match our internal linting guidelines.

### CODE REVIEW

File: packages/driver-dom/src/index.js [hunk 1]
```diff
@@ -14,106 +14,66 @@ const STYLE = 'style';
...
-}
-
 export function createBody() {
   return document.body;
 }
 
-export function createComment(content) {
-  return document.createComment(content);
+export function createEmpty(component) {
+  return document.createComment(' _ ');
```
Comment (wssgcg1213) [inline packages/driver-dom/src/index.js]: why not empty string?

File: packages/driver-server/src/index.js [hunk 2]
```diff
@@ -198,13 +197,13 @@ const Driver = {
...
-    for (let key in styles) {
-      node.style[key] = styles[key];
+  setStyle(node, style) {
+    for (let key in style) {
+      node.style[key] = style[key];
     }
   },
 
-  setNativeProps(node, props, skipSetStyles) {
+  setNativeProps(node, props, skipsetStyle) {
```
Comment (wssgcg1213) [inline packages/driver-server/src/index.js]: The second `s` character should be uppercased.

File: packages/driver-server/src/index.js [hunk 3]
```diff
@@ -213,10 +212,10 @@ const Driver = {
 
       if (value != null) {
         if (prop === STYLE) {
-          if (skipSetStyles) {
+          if (skipsetStyle) {
             continue;
           }
-          this.setStyles(node, value);
+          this.setStyle(node, value);
         } else if (EVENT_PREFIX_REGEXP.test(prop)) {
```
Comment (wssgcg1213) [inline packages/driver-server/src/index.js]: Tip: `EVENT_PREFIX_REGEXP.test(prop)` can be cached to get performance upgrade.

File: packages/rax/src/vdom/empty.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ import BaseComponent from './base';
  */
 class EmptyComponent extends BaseComponent {
   createNativeNode() {
-    return Host.driver.createEmpty();
+    return Host.driver.createEmpty(this);
```
Comment (wssgcg1213) [inline packages/rax/src/vdom/empty.js]: 这个 this 是做何考虑来使用的? 我看虽然协议上加了这个参数, 实现里并没有使用的地方, 如果仅创建empty元素的话可以不传参数的吧

File: packages/rax/src/vdom/text.js [hunk 5]
```diff
@@ -20,7 +20,7 @@ class TextComponent extends BaseComponent {
   }
 
   createNativeNode() {
-    return Host.driver.createText(this._currentElement);
+    return Host.driver.createText(this._currentElement, this);
```
Comment (wssgcg1213) [inline packages/rax/src/vdom/text.js]: 这个this也有些奇怪

File: packages/driver-dom/src/index.js (hunk 1 above)
Comment (yuanyan) [reply packages/driver-dom/src/index.js]: 已经直接修改到 ''，减少包大小，语义也没问题

File: packages/driver-server/src/index.js (hunk 2 above)
Comment (yuanyan) [reply packages/driver-server/src/index.js]: 已经修改了

File: packages/driver-server/src/index.js (hunk 3 above)
Comment (yuanyan) [reply packages/driver-server/src/index.js]: driver-server 目前只在 test-render 里会用到，暂时不考虑优化性能

File: packages/rax/src/vdom/empty.js (hunk 4 above)
Comment (yuanyan) [reply packages/rax/src/vdom/empty.js]: hybrate 模式用到这个参数

File: packages/rax-hydrate/README.md [hunk 6]
```diff
@@ -1 +1 @@
-# rax-hydrate
\ No newline at end of file
+# rax-hybrate
```
Comment (chenjun1011) [inline packages/rax-hydrate/README.md]: 这里是拼错了吗？

File: packages/rax/src/vdom/native.js [hunk 7]
```diff
@@ -439,8 +439,9 @@ class NativeComponent extends BaseComponent {
   }
 
   createNativeNode() {
-    var nativeNode = Host.driver.createElement(this._instance);
-    Instance.set(nativeNode, this._instance);
+    const instance = this._instance;
+    const nativeNode = Host.driver.createElement(instance.type, instance.props, this);
```
Comment (chenjun1011) [inline packages/rax/src/vdom/native.js]: 那 driver-worker driver-dom 里 createElement 的入参是不是也要同步改下

Comment (yuanyan) [reply packages/rax/src/vdom/native.js]: driver-worker 与driver-weex 目前都是rax 0.6.7，后续的确都需要改的

File: packages/rax-hydrate/README.md (hunk 6 above)
Comment (yuanyan) [reply packages/rax-hydrate/README.md]: 已经修改

**Pull request merged**